pkg/controller/expansion/expansion_controller.go

```diff
@@ -227,5 +227,5 @@ func setStatusError(status *v1beta1.ExpansionTemplatePodStatus, etErr error) {
 	}
 
 	e := &v1beta1.ExpansionTemplateError{Message: etErr.Error()}
-	status.Status.Errors = append(status.Status.Errors, e)
+	status.Status.Errors = []*statusv1beta1.ExpansionTemplateError{e}
```

Comment:
yea, this old code would append the error infinitely lol. An error would be appended, which would trigger a reconcile, which would append an error, and so on 😭 😭 😭 

Didn't catch it because there was no way to reach the code branch at the time (`expansionSystem.Upsert` never returned an error)